pkg/controller/healthcheck/add.go

```diff
@@ -78,6 +78,11 @@ func RegisterHealthChecks(mgr manager.Manager, opts healthcheck.DefaultAddArgs)
 				HealthCheck:   general.NewSeedDeploymentHealthChecker(aws.CSISnapshotControllerName),
 				PreCheckFunc:  csiEnabledPreCheckFunc,
 			},
+			{
+				ConditionType: string(gardencorev1beta1.ShootControlPlaneHealthy),
+				HealthCheck:   general.NewSeedDeploymentHealthChecker(aws.CSISnapshotValidation),
+				PreCheckFunc:  csiEnabledPreCheckFunc,
```

Comment:
As said, this check will fail the entire time until the `Shoot` got reconciled (which might take up to `24h`). So, you should add an additional pre-check for whether the `Deployment` exists in the shoot namespace in the seed.

Reply:
It can't be implemented using PreCheckFunc because we can't get client. Please check [here](https://github.com/gardener/gardener-extension-provider-aws/blob/55b98dbc3b29d191fef4df4eb518ad7c159b986f/vendor/github.com/gardener/gardener/extensions/pkg/controller/healthcheck/actuator.go#L58).

Reply:
You can use `mgr.GetClient()` to get it. But I see that we'd still lack the `context`. Potentially the signature of the `PreCheckFunc` can be improved?

Reply:
Yes that again make it two step process i.e. first make changes in g/g and then in extensions, also with time in future i.e. with next release this precheck need to be removed otherwise [this check](https://github.com/gardener/gardener-extension-provider-aws/blob/55b98dbc3b29d191fef4df4eb518ad7c159b986f/vendor/github.com/gardener/gardener/extensions/pkg/controller/healthcheck/general/deployment.go#L98-L103) will not be respected. I will prefer to leave it, WDYT?

Reply:
Your concerns are valid, but simply keeping it as it is now is no option, I fear. Hundreds, or rather thousands of `Shoot`s (depending on landscape size) will have failing health checks until they get reconciled which might take `24h`. This will not help operators of such landscapes.

Maybe we can be pragmatic and

1. Not add the health check for this new `Deployment` now (add it but comment it)
2. Add a TODO to enable it only in a future release
3. Still improve the `PreCheck` signature in g/g so that we can add such logic in the future in case we need it

Reply:
Ok, Thanks for explanation, 
But still we will be required to make changes two time in provider extension i.e add the precheck condition `for whether the Deployment exists` and in the next release remove the precheck condition, is there a possible solution so we can make it one step process?

Reply:
My proposal (https://github.com/gardener/gardener-extension-provider-aws/pull/507#discussion_r816566940) was to not have the health check now (1) and only enable it later (2) by just removing the commented code. There is no need to introduce the pre-check logic we discussed earlier then because we can assume that the `Deployment` exists in the meantime.
I don't think we can find a one-step approach, but the proposal is pragmatic with minimal impact.

Reply:
PTAL 5c899f2d21db9ea4c3e6c96a39ee85c3df8b93de